tools/install.py

```diff
@@ -50,12 +51,13 @@ def copy_emscripten(target):
   os.chdir(emscripten_root)
   for root, dirs, files in os.walk('.'):
     # Handle the case were the target directory is underneath emscripten_root
-    if os.path.abspath(root) == target:
+    if os.path.abspath(root) == os.path.abspath(target):
+      dirs.clear()
```

Comment:
what does this line do? right after it we continue anyhow, which means we assign a new value to `dirs`, don't we?

Reply:
Modifying the `dirs` value like this prevents recursion into subdirectories.

Reply:
..how?

```python
for x in range(0, 10):
  if condition():
    x = 5 # isn't this pointless?
    continue
```

Reply:
https://docs.python.org/3/library/os.html#os.walk:

```
When topdown is True, the caller can modify the dirnames list in-place (perhaps using del or slice assignment), and walk() will only recurse into the subdirectories whose names remain in dirnames; this can be used to prune the search, impose a specific order of visiting, or even to inform walk() about directories the caller creates or renames before it resumes walk() again. Modifying dirnames when topdown is False has no effect on the behavior of the walk, because in bottom-up mode the directories in dirnames are generated before dirpath itself is generated.
```

Reply:
Oh, wow, that's a surprising API, thanks...

Reply:
I think its different because its not primitive type, but a tuple containing lists, so the iterator function can tell if you mutate what it gives you.    In this case I've found this feature really useful in the past since without this you would needlessly descend into directories you don't care about.